### PR TITLE
DEVTOOLING-147: Refactored emergency group

### PR DESCRIPTION
**Pull request by carnellj-genesys**

### CODE REVIEW

File: genesyscloud/architect_emergencygroup/genesyscloud_architect_emergencygroup_proxy.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	for pageNum := 2; pageNum <= *emergencyGroupConfigs.PageCount; pageNum++ {
+		if getErr != nil {
+			return nil, resp, fmt.Errorf("Failed to get page of emergency group configs: %v", getErr)
+		}
+
+		if emergencyGroupConfigs.Entities == nil || len(*emergencyGroupConfigs.Entities) == 0 {
+			break
+		}
+
```
Comment (charliecon) [inline genesyscloud/architect_emergencygroup/genesyscloud_architect_emergencygroup_proxy.go:98]: Looks like we're missing some logic inside this for loop. There's no actual API call, and we're not appending the list of total records

File: genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	description := d.Get("description").(string)
+	divisionId := d.Get("division_id").(string)
+	enabled := d.Get("enabled").(bool)
+
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	ap := getArchitectEmergencyGroupProxy(sdkConfig)
+
+	diagErr := genesyscloud.RetryWhen(genesyscloud.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Get current emergency group version
+		emergencyGroup, resp, getErr := ap.getAllArchitectEmergencyGroup(ctx, d.Id())
```
Comment (charliecon) [inline genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go:133]: It's a bit confusing how the function "getAllArchitectEmergencyGroup" returns one emergency group by ID. Was this a mistake, or is it how the API works?

**Comment by charliecon** [comment]
Also, looks like `go generate` need to be ran, and the version update probably caused those merge conflicts

File: genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go (hunk 2 above)
Comment (carnellj-genesys) [reply genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go:133]: No.  I must have cut and past then forgot to remove the All.  Fixed

File: genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+		if emergencyGroup.State != nil && *emergencyGroup.State == "deleted" {
+			d.SetId("")
+			return nil
+		}
+
+		d.Set("name", *emergencyGroup.Name)
+		d.Set("division_id", *emergencyGroup.Division.Id)
+
+		if emergencyGroup.Description != nil {
```
Comment (dginty4) [inline genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go:99]: These if statements can be replaced with `resourcedata.SetNillableValue`

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Review by dginty4 — APPROVED** [review]

**Pull request merged**